### ISSUE TITLE
Docker configuration

### ISSUE THREAD

**Issue by TrevorHansen** [stp/stp#416]
I'm thinking of making three changes to the Docker configuration:
1) Building using a later version of Ubuntu (currently 16.04) to access a more modern gcc (currently it's using 5.5.0).
2) Building CMS and STP in release mode with assertions disabled.
3) Updating the CMS from 5.6.5 to 5.8.0.

If the Docker config is used for automating testing, then (2) is bad. Are there other considerations?

**Comment by rgov** [comment]
You could make a Docker build argument for enabling assertions, which we could do use in CI.

**Comment by TrevorHansen** [comment]
I inadvertently submitted some changes to the docker file in 7351d7d2179b8d70703fe8569af3d5d4902452c1

It disabled assertions and upgraded the CMS. So 2 of 3 that I wanted to make.

I'm happy to fix it as you suggest @rgov, if the CI is now broken? Do you know if it broke the CI? Or has it caused some other problems?

I'd also like to upgrade the builder to  Ubuntu to 20.04, but I get this error when I try:

```
/usr/bin/ld: attempted static link of dynamic object `/usr/lib/x86_64-linux-gnu/libboost_program_options.so.1.71.0'
collect2: error: ld returned 1 exit status
make[2]: *** [tools/stp/CMakeFiles/stp-bin.dir/build.make:104: stp] Error 1
make[1]: *** [CMakeFiles/Makefile2:1092: tools/stp/CMakeFiles/stp-bin.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

This looks similar to this:
https://github.com/meelgroup/bosphorus/issues/15

But I don't see the fix to in the associated patches, does anyone know how to fix it? 

Thanks.

**Comment by rgov** [comment]
Looks like CI does not currently use Docker so we don't need to address the build arg issue right now.

What is the compiler/linker invocation that leads to the error?

**Comment by TrevorHansen** [comment]
Editing the Dockerfile so it builds with Ubuntu 20 rather than 16 gives the error. So editing the Dockerfile:

```
diff --git a/Dockerfile b/Dockerfile
index 9b37a0b0..841436b1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04 as builder
+FROM ubuntu:20.04 as builder
 
 LABEL maintainer="Mate Soos"
 LABEL version="5.0"
```

Then running:
`docker built -t stp .
`
Gives me the linker error because the .so rather than the .a is linked to.

**Comment by TrevorHansen** [comment]
Thanks @rgov . You fixed this with 4ab147caa4b67fc5de0a291ee9852afd91642b3b
